swarm/storage/resource_test.go

```diff
@@ -262,6 +264,133 @@ func TestResourceHandler(t *testing.T) {
 
 }
 
+func TestResourceMultihash(t *testing.T) {
+
+	// signer containing private key
+	signer, err := newTestSigner()
+	if err != nil {
+		t.Fatal(err)
+	}
+	validator := newTestValidator(signer.signContent)
+
+	// make fake backend, set up rpc and create resourcehandler
+	backend := &fakeBackend{
+		blocknumber: int64(startBlock),
+	}
+
+	// set up rpc and create resourcehandler
+	rh, datadir, _, teardownTest, err := setupTest(backend, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer teardownTest()
```

Comment:
`teardownTest()` should be right after the `setupTest`, call, otherwise it will not run if setupTest gives back an error.

Reply:
`setupTest` will only return `Error` if it can't make tempdir, in which case the `teardownTest` will be nil.